### PR TITLE
Added Visit Dao Test class for Mock unit test -- WIP

### PR DESCRIPTION
**Pull request by Mellorison**

### CODE REVIEW

File: core/src/test/java/zm/gov/moh/core/VisitDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Mock
+    private Database db;
+
+    @Spy
+    private VisitDao visitDao;
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        long[] pushedEntityIDs = db.entityMetadataDao().findEntityIdByTypeRemoteStatus(EntityType.VISIT.getId(), RemoteWorker.Status.PUSHED.getCode());
```
Comment (AndreLesa) [inline core/src/test/java/zm/gov/moh/core/VisitDaoTest.java]: The setup should be inserting things into the Database and such, not querying them.

File: core/src/test/java/zm/gov/moh/core/VisitDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+
+    @Test
+    public void findEntityNotWithIDTest() {
+       when(db.visitDao().findEntityNotWithId( Constant.LOCAL_ENTITY_ID_OFFSET, db.entityMetadataDao().findEntityIdByTypeRemoteStatus(EntityType.VISIT.getId(), RemoteWorker.Status.PUSHED.getCode()))).thenReturn(longEntitiesWithoutIDs);
+        visitDao.findEntityNotWithId(Constant.LOCAL_ENTITY_ID_OFFSET, db.entityMetadataDao().findEntityIdByTypeRemoteStatus(EntityType.VISIT.getId(), RemoteWorker.Status.PUSHED.getCode()));
+        when(visitDao.getMaxId()).thenReturn(offset);
+        visitDao.getMaxId();
+        System.out.print("EntityWithoutID was tested successfully");
```
Comment (AndreLesa) [inline core/src/test/java/zm/gov/moh/core/VisitDaoTest.java]: This is not a test, test should use the Framework's "Test" such as assertEqual, assertFalse e.t.c
That's the actual test, everything else is just done to get to the asserts.

I would much rather have a few tests done correctly with asserts that either pass or fail than a lot of tests that are not actually testing.

**Review by AndreLesa — CHANGES_REQUESTED** [review]

File: core/src/test/java/zm/gov/moh/core/VisitDaoTest.java (hunk 2 above)
Comment (Mellorison) [reply core/src/test/java/zm/gov/moh/core/VisitDaoTest.java]: This is WIP. But Well noted and Understood.

File: core/src/test/java/zm/gov/moh/core/VisitDaoTest.java (hunk 1 above)
Comment (Mellorison) [reply core/src/test/java/zm/gov/moh/core/VisitDaoTest.java]: Noted. A Mocked List of mocked data as requested. We are on the verge of implementation...

**Pull request merged**